### PR TITLE
[FLINK-21290][table-planner] Support project push down for window TVF.

### PR DESCRIPTION
**Pull request by beyond1920**
## What is the purpose of the change

This pull request aims to support project push down for window TVF.

## Brief change log

  - Introduce `ProjectWindowTableFunctionTransposeRule`
  - Update xml and .out of effected test case.

## Verifying this change
  - Existed UT

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by beyond1920** [comment]
@godfreyhe Thanks for review, I've updated, please have a look.

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/ProjectWindowTableFunctionTransposeRuleTest.scala [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+/**
+ * Test for [[ProjectWindowTableFunctionTransposeRule]].
+ */
+class ProjectWindowTableFunctionTransposeRuleTest extends TableTestBase {
+  private val util = streamTestUtil()
+
+  @Before
+  def setup(): Unit = {
+    util.buildStreamProgram(FlinkStreamProgram.LOGICAL)
```
Comment (godfreyhe) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/ProjectWindowTableFunctionTransposeRuleTest.scala:34]: we should rebuild the program which contains the necessary rules only

**Review by godfreyhe — APPROVED** [review]
LGTM

**Comment by beyond1920** [comment]
@flinkbot run azure

**Comment by beyond1920** [comment]
@flinkbot run travis

**Comment by beyond1920** [comment]
@flinkbot run azure

**Comment by beyond1920** [comment]
https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=22300&view=results

**Comment by beyond1920** [comment]
@flinkbot run azure

**Pull request closed without merging**